aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -323,6 +323,9 @@ Tensor _histc_cuda_template(
     maxvalue = maxvalue + 1;
   }
 
+  TORCH_CHECK(!(std::isinf(minvalue) || std::isinf(maxvalue) || std::isnan(minvalue) || std::isnan(maxvalue)), "range of [", minvalue, ", ", maxvalue, "] is not finite");
+  TORCH_CHECK(minvalue < maxvalue, "max must be larger than min");
```

Comment:
Windows CI gives error:
```
15:26:54 C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt\corecrt_math.h(409): error: more than one instance of overloaded function "fpclassify" matches the argument list:
15:26:54 SummaryOps.cu
15:26:54             function "fpclassify(float)"
15:26:54             function "fpclassify(double)"
15:26:54             function "fpclassify(long double)"
15:26:54             argument types are: (uint8_t)
15:26:54           detected during:
15:26:54             instantiation of "__nv_bool isinf(_Ty) [with _Ty=uint8_t]" 
15:26:54 C:/Jenkins/workspace/pytorch-builds/pytorch-win-ws2016-cuda9-cudnn7-py3-build/aten/src/ATen/native/cuda/SummaryOps.cu(326): here
15:26:54             instantiation of "at::Tensor at::<unnamed>::_histc_cuda_template(const at::Tensor &, int64_t, input_t, input_t) [with input_t=uint8_t]" 
15:26:54 C:/Jenkins/workspace/pytorch-builds/pytorch-win-ws2016-cuda9-cudnn7-py3-build/aten/src/ATen/native/cuda/SummaryOps.cu(357): here
```
To fix this issue, in this file I think we should write `THCNumerics<input_t>::isinf(val)` instead of `std::isinf(val)`, and `THCNumerics<input_t>::isnan(val)` instead of `std::isnan(val)` (this is taken from https://github.com/pytorch/pytorch/blob/243298668caca7ee7de454ebc4d99c487ec9b272/aten/src/ATen/native/cuda/MultinomialKernel.cu#L244-L245
). This only applies to CUDA code though, and we don't need to change the CPU code.